pkg/providers/tinkerbell/validate.go

```diff
@@ -3,17 +3,32 @@ package tinkerbell
 import (
 	"errors"
 	"fmt"
+	"net/url"
 	"strings"
 
 	tinkv1alpha1 "github.com/tinkerbell/tink/pkg/apis/core/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/logger"
 	"github.com/aws/eks-anywhere/pkg/networkutils"
 	"github.com/aws/eks-anywhere/pkg/providers/tinkerbell/hardware"
 )
 
 func validateDatacenterConfig(config *v1alpha1.TinkerbellDatacenterConfig) error {
+	if config.Spec.OSImageURL != "" {
+		if _, err := url.ParseRequestURI(config.Spec.OSImageURL); err != nil {
+			return fmt.Errorf("parsing osImageOverride: %v", err)
+		}
+	}
+
+	if config.Spec.HookImagesURLPath != "" {
+		if _, err := url.ParseRequestURI(config.Spec.HookImagesURLPath); err != nil {
+			return fmt.Errorf("parsing hookOverride: %v", err)
+		}
+		logger.Info("hook path override set", "path", config.Spec.HookImagesURLPath)
```

Comment:
If we going to call this from a webhook, can we use logger? Or is it better to just return errors?

Reply:
I see [this](https://github.com/aws/eks-anywhere/blob/74af3394729dab77473e9435f254a0f708572abe/pkg/api/v1alpha1/vspheremachineconfig.go#L123) mutation webhook using logger, so I can keep it for now, and then remove it (if reqd) during the webhook implementation by testing it.